### PR TITLE
{WIP}(GH-598) add outputdirectory for choco pack

### LINKED ISSUE #598: Add possibility to set an output folder for choco pack
Like [`nuget pack`](https://docs.nuget.org/consume/command-line-reference#pack-command) it could be useful to have the `-OutputDirectory` option.

### PR DESCRIPTION
**Pull request by laedit**
Fix #598 
Add a new option outputdirectory for PackCommand and the corresponding
OutputDirectory property in ChocolateyConfiguration.

Pack run can't be tested because PackageBuilder isn't mocked and not
use mocked filesystem.

Should I add an indirection for PackageBuilder through an IPackageBuilderFactory which can be passed to NugetService constructor?

### CODE REVIEW

File: src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs [hunk 1]
```diff
@@ -186,6 +186,7 @@ private void append_output(StringBuilder propertyValues, string append)
         public string Version { get; set; }
         public bool AllVersions { get; set; }
         public bool SkipPackageInstallProvider { get; set; }
+        public string OutputDirectory { get; set; }
```
Comment (ferventcoder) [inline src/chocolatey/infrastructure.app/configuration/ChocolateyConfiguration.cs]: :+1:

**Comment by ferventcoder** [comment]
@laedit do you still consider this a WIP? or is it ready to go?

**Comment by laedit** [comment]
It works, but I haven't find a way to add unit test for the pack run because the `PackageBuilder` isn't mocked and do not use a mocked filesystem.

So if you are ok with that, the PR is ready to go, otherwise I will add an indirection for PackageBuilder through an IPackageBuilderFactory which can be passed to NugetService constructor.
Unless there is a better way to do that?

**Comment by ferventcoder** [comment]
Yeah, I've found certain things are very difficult to mock, although I've heard with all the abstractions in NuGet v3, that has been helped. We will be upgrading to that at some point, hoping this year (depending on where it prioritizes). 

I think it is fine to forego needing that if everything checks out. You could add a scenario for this in integration specs. I could also do that. 👍

**Comment by laedit** [comment]
I will do that and clean the unit tests.
It's in `chocolatey.tests.integration`?
I don't see a `PackScenario`, do I just add one or do you prefer that I do otherwise?

**Comment by ferventcoder** [comment]
Add one.

**Comment by gep13** [comment]
@ferventcoder if/when this gets accepted, i will need to update the Cake Addin for Chocolatey to make use of it :smile: :+1:

**Comment by laedit** [comment]
I have added the integration tests and done a rebase since the branch was behind.
Sorry for the mess.

**Comment by ferventcoder** [comment]
Please don't merge, always rebase. Right now this is not in a state that it can even be reviewed. You can try to back out the merge and just provide the commits for the changes or you can close and reopen this with just the changes.

**Comment by laedit** [comment]
Sorry about that, I'll try to fix it this week.

**Comment by laedit** [comment]
I am sorry, this one is irrecoverable, I had to create a new PR #891.

**Pull request closed without merging**